pkg/cmd/connector/create/create.go

```diff
@@ -114,3 +133,175 @@ func runCreate(opts *options) error {
 
 	return nil
 }
+
+func createServiceAccount(opts *factory.Factory, shortDescription string) (*kafkamgmtclient.ServiceAccount, error) {
+	conn, err := opts.Connection()
+	if err != nil {
+		return nil, err
+	}
+	serviceAccountPayload := kafkamgmtclient.ServiceAccountRequest{Name: shortDescription}
+
+	serviceacct, httpRes, err := conn.API().
+		ServiceAccountMgmt().
+		CreateServiceAccount(opts.Context).
+		ServiceAccountRequest(serviceAccountPayload).
+		Execute()
+
+	if httpRes != nil {
+		defer httpRes.Body.Close()
+	}
+
+	if err != nil {
+		return nil, err
+	}
+	opts.Logger.Info(opts.Localizer.MustLocalize("connector.sa.created",
+		localize.NewEntry("ClientId", serviceacct.ClientId), localize.NewEntry("ClientSecret", serviceacct.ClientSecret)))
+	return &serviceacct, nil
+}
+
+func setDefaultValuesFromFlags(connector *connectormgmtclient.ConnectorRequest, opts *options) error {
+	if opts.kafkaId != "" {
+		connector.Kafka = connectormgmtclient.KafkaConnectionSettings{
+			Id: opts.kafkaId,
+		}
+	}
+
+	if opts.namespace != "" {
+		connector.NamespaceId = opts.namespace
+	}
+
+	if opts.name != "" {
+		connector.Name = opts.name
+	}
+
+	if opts.serviceAccount {
+		serviceAccount, err1 := createServiceAccount(opts.f, "connector-"+connector.Name)
```

Comment:
This will be problematic if the connector name has underscore. As service account description can't have underscore.
```
 > ./rhoas connector create --file "rama-connector.json" --name rama_connector --create-service-account -v

{"reason":"name does not match ^[a-z]([-a-z0-9]*[a-z0-9])?$","operation_id":"cbogqu2khkqj0qi9hpp0","id":"38","kind":"Error","href":"/api/kafkas_mgmt/v1/errors/38","code":"KAFKAS-MGMT-38"}
```